.github/workflows/remove_deprecated_decls.yml

```diff
@@ -0,0 +1,207 @@
+        if [[ -n "${RAW_FROM_DATE:-}" ]]; then
+          echo "Processing raw_from_date: '${RAW_FROM_DATE}'"
+
+          if from_date=$(validate_iso_date "${RAW_FROM_DATE}"); then
+            echo "Valid ISO date format detected"
+          else
+            echo "Not a valid ISO date, trying date parser..."
+            from_date=$(parse_date_safely "${RAW_FROM_DATE}")
+          fi
+        else
+          echo "No raw_from_date provided, using 5 years ago"
+          from_date=$(date -d '5 years ago' +%Y-%m-%d)
+        fi
+
+        # Process to_date
+        if [[ -n "${RAW_TO_DATE:-}" ]]; then
+          echo "Processing raw_to_date: '${RAW_TO_DATE}'"
+
+          if to_date=$(validate_iso_date "${RAW_TO_DATE}"); then
+            echo "Valid ISO date format detected"
+          else
+            echo "Not a valid ISO date, trying date parser..."
+            to_date=$(parse_date_safely "${RAW_TO_DATE}")
+          fi
+        else
+          echo "No raw_to_date provided, using 6 months ago"
+          to_date=$(date -d '6 months ago' +%Y-%m-%d)
+        fi
+
+        echo "Parsed dates - from: $from_date, to: $to_date"
+
+        # Validate date range - from_date should not be after to_date
+        if [[ "$from_date" > "$to_date" ]]; then
+          echo "Error: from_date ($from_date) cannot be after to_date ($to_date)" >&2
+          exit 1
+        fi
+
+        # Set outputs
+        echo "from_date=$from_date" >> "$GITHUB_OUTPUT"
+        echo "to_date=$to_date" >> "$GITHUB_OUTPUT"
+
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Configure Lean
+      uses: leanprover/lean-action@f807b338d95de7813c5c50d018f1c23c9b93b4ec # 2025-04-24
+      with:
+        auto-config: false
+        use-github-cache: false
+        use-mathlib-cache: false
+
+    - name: Retrieve caches for unmodified repositories
+      run: |
+        # Retrieve the cache for the unmodified repository
+        # show the current pwd:
+        printf $'Current directory is: %s\n' "$(pwd)"
+        printf $'::group::Running lake env\n'
+        lake env | tr ':' '\n'
+        echo "::endgroup::"
+        for repo in Mathlib Archive Counterexamples; do
+          echo "::group::Retrieving the cache for ${repo}"
+          lake exe cache get "$repo" || true
+          echo "::endgroup::"
+        done
+
+    - name: set up deprecated doubles
+      env:
+        OLD_DATE: "${{ steps.process_dates.outputs.from_date }}"
+        NEW_DATE: "${{ steps.process_dates.outputs.to_date }}"
+        DRY_RUN: "${{ toJson(inputs.dry_run) }}"
+      run: |
+        # We create a temporary file to run the command that adds the set_option linter.removeDeprecations
+        # to the relevant files and saves the modified files with a new name
+        tmplean="$(mktemp -p Mathlib --suffix=.lean RMDXXX)"
+        echo "::group::Creating ${tmplean} file"
+        {
+          for repo in Mathlib Archive Counterexamples; do
```

Comment:
Maybe we should comment out `Archive` and `Counterexamples` since the script isn't able to remove them at the moment?